### ISSUE TITLE
'className' instead of 'class' within JSX markup

### ISSUE THREAD

**Issue by dexterlohnes** [yannickcr/eslint-plugin-react#186]
Hi,

Is there anyway we could add a rule that lets you lint for the use of the 'class' field when it should be 'className'?

ex:
```javascript
render () {
	console.log('Render');
    return (
		<div key={this.props.id} class='bookListItem'>
			<p>{this.props.title}</p>
			<p>{this.props.description}</p>
		</div>
    );
  }
```

**Comment by yannickcr** [comment]
Hi, the [no-unknown-property](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unknown-property.md) rule handle this. It also warn you for the other HTML attributes that are different in JSX (like `for` -> `htmlFor`, `colspan` -> `colSpan`, etc.).

**Comment by greim** [comment]
Sometimes custom elements pass `className` through to a DOM output element. For example react-router's `<Link/>`:

```html
<Link class="foo"></Link>
```

This is an error, but `no-unknown-property` won't catch it because it isn't a DOM element. Is there a blanket way to flag all instances of `class=` since it's almost always an error?

**Comment by lencioni** [comment]
@greim I think you can use the forbid-component-props rule to prevent whatever props you like on your components. https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-component-props.md

**Comment by greim** [comment]
@lencioni This works, thank you!

**Comment by Primajin** [comment]
Sorry for digging up this old issue here, but is there a way to lint against `no-known-property`?

Bear with me for a second: What I mean is when people pass props down to custom components, but use attribute names, that are actually native HTML ones, like `id`,`title`,`name`, `disabled` and so on. There are global ones and ones that can only be used on specific elements. [See List](https://developer.mozilla.org/docs/Web/HTML/Attributes)

```jsx
<p title="See this when you hover me">Everything is fine here</p>
```
vs.
```jsx
import P from '../myCustomComponent';
...
<P title="See this when you hover me">Is this an attribute or prop?</P>
```
Expected
```jsx
import P from '../myCustomComponent';
...
<P hoverTitle="See this when you hover me">This is better</P>
```

Thanks for consideration.

**Comment by ljharb** [comment]
Yes, if you want a rule to forbid invalid html props based on the element type, we’d need a different rule.

**Comment by Primajin** [comment]
Cool, opened #1990
